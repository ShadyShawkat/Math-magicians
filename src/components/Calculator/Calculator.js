/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import Button from '../Button/Button';

import styles from './Claculator.module.css';

const btnTexts = [
  'AC',
  '+/-',
  '%',
  'รท',
  '7',
  '8',
  '9',
  'x',
  '4',
  '5',
  '6',
  '-',
  '1',
  '2',
  '3',
  '+',
  '0',
  '.',
  '=',
];

class Calculator extends React.Component {
  render() {
    return (
      <div className={styles.calculator}>
        <input className={styles.calculator_output} value="0" />
        {btnTexts.map((btnTxt) => (
          <Button
            key={Math.random()}
            isZero={btnTxt === '0'}
            isOperator={['รท', 'x', '-', '+', '='].includes(btnTxt)}
          >
            {btnTxt}
          </Button>
        ))}
      </div>
    );
  }
}

export default Calculator;
