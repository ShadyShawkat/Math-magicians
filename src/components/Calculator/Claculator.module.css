.calculator {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr;
  height: 100%;
  width: 100%;
}

.calculator > * {
  border: 2px solid #bababa;
}

.calculator_output {
  all: unset;
  text-align: right;
  grid-column: 1/5;
  background-color: #858694;
  color: white;
  font-size: 3rem;
  padding-right: 1rem;
}
