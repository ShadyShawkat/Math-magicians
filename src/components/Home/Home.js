import React from 'react';
import styles from './Home.module.css';

const Home = () => (
  <div className={styles.home}>
    <h2>Welcome to my page!</h2>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industrys standard dummy text ever since the
      1500s, when an unknown printer took a galley of type and scrambled it to
      make a type specimen book. It has survived not only five centuries, but
      also the leap into electronic typesetting, remaining essentially
      unchanged. It was popularised in the 1960s with the release of Letraset
      sheets
    </p>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industrys standard dummy text ever since the
      1500s, when an unknown printer took a galley of type and scrambled it to
      make a type specimen book. It has survived not only five centuries
    </p>
  </div>
);

export default Home;
