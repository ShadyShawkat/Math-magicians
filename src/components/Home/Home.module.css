.home {
  font-size: 1.5em;
}

.home h2 {
  margin-bottom: 1em;
  font-size: 1.5em;
}

.home p {
  margin-top: 0.5em;
  font-weight: lighter;
  font-size: 0.8em;
}
