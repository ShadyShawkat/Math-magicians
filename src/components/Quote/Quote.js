import React from 'react';
import styles from './Quote.module.css';

const Quote = () => (
  <div className={styles.quote}>
    <div className={styles['quote-text']}>
      &quot;Mathmatics is not about numbers, equation, computations, or
      algorithms: it is about understanding.&quot;
    </div>
    <br />
    <span>-William Paul Thurston</span>
  </div>
);

export default Quote;
