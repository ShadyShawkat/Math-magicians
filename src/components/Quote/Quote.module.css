@import url('https://fonts.googleapis.com/css2?family=Pacifico&display=swap');

.quote {
  margin-top: 2em;
  font-size: 2em;
}

.quote-text {
  font-family: 'Pacifico', cursive;
}

.quote span {
  float: right;
}
