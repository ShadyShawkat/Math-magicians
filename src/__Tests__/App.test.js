import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Home from '../components/Home/Home';
import Quote from '../components/Quote/Quote';
import Calculator from '../components/Calculator/Calculator';

describe('Testing components render', () => {
  it('Tests Home.js component render', () => {
    const tree = render(<Home />);
    expect(tree).toMatchSnapshot();
  });
  it('Tests Quote.js component render', () => {
    const tree = render(<Quote />);
    expect(tree).toMatchSnapshot();
  });
  it('Tests Calculator.js component render', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
    const output = screen.getByTestId('output');
    userEvent.click(screen.getByText('1'));
    expect(output).toHaveValue('1');
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('='));
    expect(output).toHaveValue('4');
  });
});
