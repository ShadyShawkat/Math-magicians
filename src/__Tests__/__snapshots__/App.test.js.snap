// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components render Tests Calculator.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator"
      >
        <input
          class="calculator_output"
          data-testid="output"
          disabled=""
          value="0"
        />
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          AC
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          +/-
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          %
        </button>
        <button
          class="
          btn 
           
          operator
        "
          type="button"
        >
          รท
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          7
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          8
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          9
        </button>
        <button
          class="
          btn 
           
          operator
        "
          type="button"
        >
          x
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          4
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          5
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          6
        </button>
        <button
          class="
          btn 
           
          operator
        "
          type="button"
        >
          -
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          1
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          2
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          3
        </button>
        <button
          class="
          btn 
           
          operator
        "
          type="button"
        >
          +
        </button>
        <button
          class="
          btn 
          zero-num 
          
        "
          type="button"
        >
          0
        </button>
        <button
          class="
          btn 
           
          
        "
          type="button"
        >
          .
        </button>
        <button
          class="
          btn 
           
          operator
        "
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator"
    >
      <input
        class="calculator_output"
        data-testid="output"
        disabled=""
        value="0"
      />
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        AC
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        +/-
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        %
      </button>
      <button
        class="
          btn 
           
          operator
        "
        type="button"
      >
        รท
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        7
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        8
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        9
      </button>
      <button
        class="
          btn 
           
          operator
        "
        type="button"
      >
        x
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        4
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        5
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        6
      </button>
      <button
        class="
          btn 
           
          operator
        "
        type="button"
      >
        -
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        1
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        2
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        3
      </button>
      <button
        class="
          btn 
           
          operator
        "
        type="button"
      >
        +
      </button>
      <button
        class="
          btn 
          zero-num 
          
        "
        type="button"
      >
        0
      </button>
      <button
        class="
          btn 
           
          
        "
        type="button"
      >
        .
      </button>
      <button
        class="
          btn 
           
          operator
        "
        type="button"
      >
        =
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing components render Tests Home.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <h2>
          Welcome to my page!
        </h2>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets
        </p>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <h2>
        Welcome to my page!
      </h2>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets
      </p>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing components render Tests Quote.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quote"
      >
        <div
          class="quote-text"
        >
          "Mathmatics is not about numbers, equation, computations, or algorithms: it is about understanding."
        </div>
        <br />
        <span>
          -William Paul Thurston
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quote"
    >
      <div
        class="quote-text"
      >
        "Mathmatics is not about numbers, equation, computations, or algorithms: it is about understanding."
      </div>
      <br />
      <span>
        -William Paul Thurston
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
